### PR TITLE
Address turn-based submission issues

### PR DESCRIPTION
**Pull request by stephencelis**
This should address the early forfeit errors that trickle in.

### CODE REVIEW

File: Sources/AppFeature/GameCenterCore.swift [hunk 1]
```diff
@@ -26,18 +26,17 @@ extension Reducer where State == AppState, Action == AppAction, Environment == A
...
+                metadata: .init(playerIndexToId: [:])
               )
               let game = GameState(
                 cubes: environment.dictionary.randomCubes(.en),
                 gameContext: .turnBased(context),
-                gameCurrentTime: date,
+                gameCurrentTime: environment.mainRunLoop.now.date,
                 gameMode: .unlimited,
-                gameStartTime: date
+                gameStartTime: match.creationDate
```
Comment (stephencelis) [inline Sources/AppFeature/GameCenterCore.swift:39]: Not that we use the game start time for much right now, but noticed this inconsistency.

File: Sources/ClientModels/TurnBasedMatchData.swift [hunk 2]
```diff
@@ -28,7 +28,7 @@ public struct TurnBasedMatchData: Codable, Equatable {
     public var playerIndexToId: [SharedModels.Move.PlayerIndex: SharedModels.Player.Id]
 
     public init(
-      playerIndexToId: [SharedModels.Move.PlayerIndex: SharedModels.Player.Id] = [:]
+      playerIndexToId: [SharedModels.Move.PlayerIndex: SharedModels.Player.Id]
```
Comment (stephencelis) [inline Sources/ClientModels/TurnBasedMatchData.swift:31]: Figure it's better to be explicit so we don't accidentally clear out old mappings.

File: Sources/GameCore/GameCore.swift [hunk 3]
```diff
@@ -1084,8 +1084,19 @@ extension Reducer where State == GameState, Action == GameAction, Environment ==
         .fireAndForget()
 
     case let .gameCenter(.turnBasedMatchResponse(.success(match))):
-      // TODO: Should this replace `GameState`?
-      state.turnBasedContext?.match = match
```
Comment (stephencelis) [inline Sources/GameCore/GameCore.swift:1088]: This was the culprit. If you play your first turn in a game and immediately quit/forfeit, the submitted game had stale match metadata.

File: Sources/GameCore/GameCore.swift [hunk 4]
```diff
@@ -1084,8 +1084,19 @@ extension Reducer where State == GameState, Action == GameAction, Environment ==
...
+      else { return .none }
+
+      var gameState = GameState(
+        gameCurrentTime: environment.mainRunLoop.now.date,
+        localPlayer: environment.gameCenter.localPlayer.localPlayer(),
+        turnBasedMatch: match,
+        turnBasedMatchData: turnBasedMatchData
+      )
+      gameState.activeGames = state.activeGames
+      gameState.isGameLoaded = true
```
Comment (stephencelis) [inline Sources/GameCore/GameCore.swift:1098]:
```suggestion
      gameState.isGameLoaded = state.isGameLoaded
```

File: Tests/AppFeatureTests/TurnBasedTests.swift [hunk 5]
```diff
@@ -83,21 +85,36 @@ class TurnBasedTests: XCTestCase {
...
-          gameContext: .turnBased(.init(localPlayer: .mock, match: .new, metadata: .init())),
-          gameMode: .unlimited,
-          gameStartTime: self.mainRunLoop.now.date,
-          moves: [],
-          secondsPlayed: 0
-        )
-      )
+      $0.game = initialGameState
+    }
+    store.send(.currentGame(.game(.onAppear)))
```
Comment (stephencelis) [inline Tests/AppFeatureTests/TurnBasedTests.swift:107]: Beefed this test up to better reflect the reality of entering a game.

**Review by mbrandonw — APPROVED** [review]

**Pull request merged**